otwcode/otwarchive
# AO3-5287 Update filter counts on a schedule.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5287

## Purpose

This PR changes the way that FilterCounts are handled. Instead of immediately updating the filter count whenever a work's tags are changed, filters needing a recount are added to a REDIS set. Once an hour, a worker will split up that REDIS set into batches of size 1000, and then workers will process each batch to recalculate the counts for the filters in the batch.

## Testing

1. Pick a medium and go to its page.
2. Pick a canonical fandom in that medium.
3. Write down the current count associated with that fandom.
4. Create a new work in that fandom.
5. Wait 1 hour.
6. Check the medium's page again and verify that the count has increased by 1.

You can also do similar tests with other actions that should change the count: e.g. deleting a work, adding or removing a canonical tag from an existing work, wrangling a tag, etc.

## References

I included #3060 as a part of this PR.
